asmrun/startup.c

```diff
@@ -133,11 +132,13 @@ void caml_main(char **argv)
   caml_debugger_init (); /* force debugger.o stub to be linked */
   exe_name = argv[0];
   if (exe_name == NULL) exe_name = "";
-  if (caml_executable_name(proc_self_exe, sizeof(proc_self_exe)) == 0)
+  proc_self_exe = caml_executable_name();
+  if (proc_self_exe != NULL)
     exe_name = proc_self_exe;
   else
     exe_name = caml_search_exe_in_path(exe_name);
   caml_sys_init(exe_name, argv);
+  caml_stat_free(exe_name);
```

Comment:
Indeed, caml_sys_init doesn't make a copy of `exe_name`, I missed that.  Will fix later.